src/test/java/com/iota/iri/network/impl/TransactionRequesterWorkerImplTest.java

```diff
@@ -0,0 +1,142 @@
+package com.iota.iri.network.impl;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import static org.mockito.Mockito.*;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.Random;
+
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import com.iota.iri.controllers.TipsViewModel;
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.controllers.TransactionViewModelTest;
+import com.iota.iri.model.Hash;
+import com.iota.iri.model.persistables.Transaction;
+import com.iota.iri.network.Node;
+import com.iota.iri.network.TransactionRequester;
+import com.iota.iri.network.impl.TransactionRequesterWorkerImpl;
+import com.iota.iri.storage.Tangle;
+import com.iota.iri.utils.Converter;
+
+public class TransactionRequesterWorkerImplTest {
+    
+    //Good
+    private static final TransactionViewModel TVMRandomNull = new TransactionViewModel(
+            getRandomTransaction(new Random()), Hash.NULL_HASH);
+    private static final TransactionViewModel TVMRandomNotNull = new TransactionViewModel(
+            getRandomTransaction(new Random()), TransactionViewModelTest.getRandomTransactionHash());
+    private static final TransactionViewModel TVMAll9Null = new TransactionViewModel(
+            get9Transaction(), Hash.NULL_HASH);
+    
+    //Bad
+    private static final TransactionViewModel TVMAll9NotNull = new TransactionViewModel(
+            get9Transaction(), TransactionViewModelTest.getRandomTransactionHash()); 
+    
+    private static Tangle tangle;
+    private static Node node;
+    private static TipsViewModel tvm;
+    
+    private static TransactionRequester requester;
+    private static TransactionRequesterWorkerImpl worker;
+    
+    @Rule 
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+    
+    @BeforeClass
+    public static void setUp(){
+        requester = mock(TransactionRequester.class);
+        requester.init(0); //Dont drop anything
+    }
+
+    @Before
+    public void before() throws NoSuchFieldException, SecurityException, Exception {
+        tangle = mock(Tangle.class);
+        tvm = mock(TipsViewModel.class);
+        
+        worker = new TransactionRequesterWorkerImpl();
+        
+        node = mock(Node.class);
+        when(node.getNeighbors()).thenReturn(new LinkedList<>());
+        
+        //Set start queue to 0 for mocking
+        setFinalStatic(worker.getClass().getDeclaredField("REQUESTER_THREAD_ACTIVATION_THRESHOLD"), 0);
```

Comment:
I actually don't like that because you are changing the behavior of the class to fit your tests.
I would rather to see a loop creating 50 dummy transactions. 

The reason is that if someone would change the value of the constant the test will pick it up easily

This is a change request